moveit_simple/include/moveit_simple/moveit_simple.h

```diff
@@ -352,4 +360,47 @@ class CartTrajectoryPoint : public TrajectoryPoint
   Eigen::Affine3d pose_;
 };
 
+class addTrajPointException: public std::runtime_error
```

Comment:
Can this be made more simple?  Exceptions should indicate root cause and not be tied to code structure.  Creating a hierarchy of exceptions for such a simple class is another indicator that it's too complex.  Think about this from the user/developer point of view.  I called a specific method or set of methods and it failed?  Why?  Here are the reasons I can think of:
 - invalid name (trajectory or point)
 - bad conversion or perhaps more specifically we couldn't convert a cartesian point to a joint point due to an IK failure, so maybe this is an IK failure
 - move execution fail (something in the sendAndWait method failed)

Can you think of any other root causes?  The number of exception classes should reflect the number of root causes.  All exceptions should inherit from `std::runtime_error`.  More complex inheritance (not just in the case of exceptions, but all code) is almost never needed.